controllers/datadogagent/feature/apm/feature.go

```diff
@@ -0,0 +1,196 @@
+			}
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: apiutils.NewBoolPointer(true),
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+					apicommonv1.TraceAgentContainerName,
+				},
+			},
+		}
+	}
+
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *apmFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+	apm := dda.Spec.Agent.Apm
+	if apiutils.BoolValue(apm.Enabled) {
+		f.hostPortEnabled = true
+		f.hostPortHostPort = *apm.HostPort
+		if apiutils.BoolValue(apm.UnixDomainSocket.Enabled) {
+			f.udsEnabled = true
+			if apm.UnixDomainSocket.HostFilepath != nil {
+				f.udsHostFilepath = *apm.UnixDomainSocket.HostFilepath
+			}
+		}
+
+		if dda.Spec.Agent.LocalService != nil {
+			f.forceEnableLocalService = apiutils.BoolValue(dda.Spec.Agent.LocalService.ForceLocalServiceEnable)
+			if dda.Spec.Agent.LocalService.OverrideName != "" {
+				f.localServiceName = dda.Spec.Agent.LocalService.OverrideName
+			}
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: apiutils.NewBoolPointer(true),
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+					apicommonv1.TraceAgentContainerName,
+				},
+			},
+		}
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *apmFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
+	if component.ShouldCreateAgentLocalService(managers.Store().GetVersionInfo(), f.forceEnableLocalService) {
+		apmPort := []corev1.ServicePort{
+			{
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(int(f.hostPortHostPort)),
+				Port:       f.hostPortHostPort,
+				Name:       apicommon.APMHostPortName,
+			},
+		}
+		if f.localServiceName == "" {
+			f.localServiceName = component.GetAgentServiceName(f.owner)
+		}
+		return managers.ServiceManager().AddService(f.localServiceName, f.owner.GetNamespace(), nil, apmPort, nil)
+	}
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *apmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *apmFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.TraceAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDAPMEnabled,
+		Value: "true",
+	})
```

Comment:
in our current set up it looks like this envvar and `DD_APM_NON_LOCAL_TRAFFIC` are also in the core agent container; might not be necessary but just something to keep in mind when testing